### DISCUSSION TITLE
Supabase UI > Auth > viewType

### DISCUSSION THREAD

**Discussion by kernelsoe** [supabase/supabase#3617] [Questions] [answered]
# How could I declare type for Auth component's view prop?

Hello! 👋

I'm trying to implement Auth in typescript based on the supabase example in [next.js auth](https://github.com/supabase/supabase/blob/master/examples/nextjs-with-supabase-auth/pages/index.js)

The ts-error said
```
Type 'string' is not assignable to type 'ViewType | undefined'.ts(2322)
Auth.d.ts(15, 5): The expected type comes from property 'view' which is declared here on type 'IntrinsicAttributes & Props'
(JSX attribute) Props.view?: ViewType | undefined
```
Do I need to declare viewType in useState?
I'm pretty beginner to typescript. Please kindly explain me how's that type checking is working?

**Comment by trankhacvy** [comment, marked as answer]
Look like supabase/ui does not export the ViewType. I think you can declare your own ViewType

```
type ViewType =
  | 'sign_in'
  | 'sign_up'
  | 'forgotten_password'
  | 'magic_link'
  | 'update_password'
  
  const [authView, setAuthView] = useState<ViewType>('sign_in')
```

**Reply by kernelsoe** [reply]
Thank you so much!

Declaring types at useState solved this. 🤗
